### ISSUE TITLE
can't change sort order of uncompleted To-Dos in Active filter:  "Can't move a completed to-do"

### ISSUE THREAD

**Issue by Alys** [HabitRPG/habitica-android#967]
I've seen a couple of reports from people trying to change the sort order of their To-Dos while the "Active" filter is enabled and receiving the error message "Can't move a completed to-do". Moving the To-Do then fails.

In the instance that Amnmr (c98733cd-d0a1-4824-b6b7-9e74c9611ba4) reported, I confirmed that the task they were trying to move and the task they were trying to move it under were not completed, so the problem is not that the app is showing completed To-Dos in the list of uncompleted ones. I also confirmed that their `tasksOrder.todos` array did not contain the IDs of any completed tasks.

**Comment by Alys** [comment]
Anmer's reports:

"why am I now getting a message that says I cannot move unfinished to do's? I used to be able to move them to put them on priority order. Thank you!"

"I'm on Android app and website. I can sync on app but couldn't find where to do it on the website, but it seems to be synced. I've ensure I'm on active on the app and on the site. I've now seen that I can sort them fine on the site and have done so, and they have gone in that order on the app now. But still can't move them on the app like I used to. The error mess says "Can't move a completed to-do. Ok" which I swear has changed now from uncompleted to-do as that makes way more sense. Either way I can't move it though. I've added the notes now like you said to one I'm trying to move on the app. Hopefully you can see that now. Thank you so much!" [those notes were just text in the tasks' Notes fields to let me identify the tasks in question]

**Comment by Alys** [comment]
Another report, from Chandler Hartley (9755792d-d13b-4c70-a4ce-274610313fc4): "I am having issues on the mobile client in moving tasks. Everytime I move job completed tasks I get a message stating I can't move completed tasks. Not sure what the issue is, but it grays over the text after. It remains unchecked though."
I've asked them to report it in-app to send debugging information.

**Comment by Alys** [comment]
Another report, from kugoi (ba58893d-a44a-47d6-9d1a-16b4ac8c58c8):
"When I try to move a todo with a due date I get a notice that I can't move completed items. On android"

Loggly shows attempts to move To-Dos that are completed, which might be something the app does in the background. I'm going to try to do some testing and watch the Loggly records for my own account.  If this bug does have a known cause already, please tell me! :)

**Comment by Alys** [comment]
I can't replicate this. Even when I deliberately move completed To-Dos, I don't see the error message saying that I can't move them (although the move does fail and they jump back to their original positions, as you'd expect). Version 1.4.3 (1975).

However I've seen enough reports of this to be sure it does happen under some circumstances.

**Comment by Irrevenant2** [comment]
Hi.

I've been having this problem. 

It seems to occur almost every single time I use the Android App to create a new To-Do and immediately try to move it.  I've taken to sorting my tasks on the website now, which works fine (albeit far too easy to accidentally drag-and-drop). 

It seems to specifically apply to brand new tasks, in my case anyway, so maybe there's a clue in there somewhere...

**Comment by Alys** [comment]
@Irrevenant2 Thanks for taking the time to tell us!

----

Sami (af7dd42e-a87e-459d-a0a8-a3f5b8b3457b) in the Report a Bug guild also has feedback about this bug:

"Hey guys, using the android app right now, and I just created a new to-do with a checklist of 5 things. It won't allow me to move this to-do down the list, and gives me the message "Can't move a completed to-do." I can do this later on the PC, but I thought I would report it anyways."

I asked: "I'd be interested to know if syncing (dragging downwards on your tasks) makes the problem go away."

Sami said: "I tried that several times, refreshing/syncing just made it impossible to move any to-do at all! (they don't get highlighted or pop out even if I long press on them) I also found that eventually none of my other to-dos could be moved as well. (same error) However, sometimes it goes back to normal and after testing the moving a few more times, it just went back to the original issue. Hope the info helps a bit."
